src/pal/src/synchmgr/synchcontrollers.cpp

```diff
@@ -578,21 +578,22 @@ namespace CorUnix
     waiting threads awakening.
     --*/
     PAL_ERROR CSynchStateController::IncrementSignalCount(
-        DWORD dwAmountToIncrement)
+        LONG lAmountToIncrement)
     {
         VALIDATEOBJECT(m_psdSynchData);
 
-        LONG lOldCount = m_psdSynchData->GetSignalCount();        
-        LONG lNewCount = lOldCount + (LONG)dwAmountToIncrement;
-
         _ASSERTE(InternalGetCurrentThread() == m_pthrOwner);
-        _ASSERT_MSG((DWORD)INT_MAX > dwAmountToIncrement,
+        _ASSERTE(lAmountToIncrement > 0);
+        _ASSERT_MSG((ULONG)INT_MAX > (ULONG)lAmountToIncrement,
             "Signal count increment %u too large (max=%u)\n",
-            dwAmountToIncrement, (DWORD)INT_MAX);
+            (ULONG)lAmountToIncrement, (ULONG)INT_MAX);
 
-        _ASSERT_MSG(lNewCount >= lOldCount && (lNewCount >= 0 && static_cast<DWORD>(lNewCount) >= dwAmountToIncrement),
+        LONG lOldCount = m_psdSynchData->GetSignalCount();
+        LONG lNewCount = lOldCount + lAmountToIncrement;
+
+        _ASSERT_MSG(lNewCount >= lOldCount && (lNewCount >= 0 && lNewCount >= lAmountToIncrement),
```

Comment:
Yea, since we know lOldCount >= 0 && lAmountToIncrement > 0, it should be sufficient to verify that lNewCount > lOldCount to catch overflow, changed it to that. The other checks here would be implied.